### ISSUE TITLE
[Bug]: `lock table bind changed` is fired for SQLs running with gorm

### ISSUE THREAD

**Issue by DanielZhangQD** [matrixorigin/matrixone#15402]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Branch Name

1.1-dev

### Commit ID

v1.1.2-444d0285d-2024-04-08

### Other Environment Information

```Markdown
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

Execute  queries in Cloud sql editor, which executes the SQLs with GORM  and the following error is fired:
```
> select 1;
Execute query error: Error 20702 (HY000): lock table bind changed
```

### Expected Behavior

Query succeeds

### Steps to Reproduce

```Markdown
See detail in Actual Behavior
```

### Additional information

It works with mysql cli

Logs: https://grafana.cn-dev.matrixone.tech/goto/eevrVZaIR?orgId=1

**Comment by DanielZhangQD** [comment]
The log starts from the upgrade of this morning:
![Uploading 企业微信截图_65982a70-34b8-4384-8e49-f39d54e963f3.png…]()
https://grafana.cn-dev.matrixone.tech/goto/l9HSYZaSR?orgId=1

**Comment by DanielZhangQD** [comment]
The same issue occurs after upgrading MO to the version `v1.1.2-1cc551d64-2024-04-09` with fix #15411:
https://grafana.cn-dev.matrixone.tech/goto/EevMA7-Sg?orgId=1

**Comment by aressu1985** [comment]
done
